cmd/snap-confine/snap-device-helper

```diff
@@ -53,8 +53,20 @@ if [ ! -e "$app_dev_cgroup" ]; then
 fi
 
 # check if it's a block or char dev
-if [ "${DEVPATH#*/block/}" != "$DEVPATH" ]; then
+# TODO: re-write this to be more robust, the bash variable substitution done 
+# here is quite awkard :-/
```

Comment:
The historical context of this is to avoid a fork/exec and to have it work with dash. I agree it is awkward. Once this is adjusted, someone may want to consider implementing the other half of https://github.com/snapcore/snapd/pull/7614 (ie, that is for snap-confine, we should have a compiled snap-device-helper that (probably) shares code with 7614 so we can avoid this shell weirdness).

Reply:
Right that all makes sense. I think it's fine until we have what was described in 7614 (a re-write not in shell that is statically compiled that doesn't need /bin/sh).